FRONTEND.md

```diff
@@ -0,0 +1,903 @@
+
+### Key Configuration Files
+
+- `package.json` - Node.js dependencies and scripts
+- `vite.config.js` - Vite build configuration for React
+- `webpack.config.js` - Webpack build configuration (legacy)
+- `tsconfig.json` - TypeScript configuration
+- `build/paths.js` - Build path configurations
+
+## Technology Selection Guidelines
+
+### React
+
+#### Use React when:
+
+- Building interactive elements that require state management and re-rendering
+- Creating components with complex user interactions
+- Developing components that are not interactive but don't exist as NLDS webcomponents
+- Building data-driven interfaces with frequent updates
+
+#### Working with React + TypeScript
+
+The project uses **Vite** as the primary build tool for React components with hot module replacement (HMR) support.
+
+**Development Commands:**
+
+- `npm run watch:react` - Start React development with file watching
+- `npm run watch:react:browser-sync` - Development with browser-sync for auto-refresh
+- `npm run bundle:react` - Build React components for production
+
+#### Webcomponents Development
+
+The project includes a **webcomponents bundle** built with Webpack for Lit-based components:
+
+**Development Commands:**
+
+- `npm run watch` - Watch mode includes webcomponents bundle
+- `npm run bundle` - Build includes webcomponents bundle
+- Visit `/webcomponents-demo/` for live component showcase
+
+**Webcomponents Architecture:**
+
+- **Lit Framework**: Modern web components with reactive properties
+- **Utrecht Integration**: Design system components for Dutch government
+- **Auto-registration**: Components register themselves via `@customElement` decorator
+- **CSP Compliant**: Uses external CSS instead of Shadow DOM for security compliance
+
+> [!NOTE]  
+> The project is transitioning from Webpack to Vite. Webpack is still used for legacy JavaScript, while Vite handles all React/TypeScript compilation.
+
+### Webcomponents
+
+The project includes both **custom Lit-based webcomponents** and **Utrecht Design System webcomponents** for building reusable, framework-agnostic UI elements.
+
+#### Use Webcomponents when:
+
+- A corresponding NLDS/Utrecht component is available
+- Building reusable, framework-agnostic components
+- Creating components that need to be shared across different parts of the application
+- Implementing standard UI patterns that benefit from encapsulation
+- Developing components that need to work with Django templates, React, and vanilla JavaScript
+
+#### Lit Framework
+
+The project uses **Lit** for building custom webcomponents with modern development patterns:
+
+- **Reactive Properties**: Automatic re-rendering when properties change
+- **CSS-in-JS**: Encapsulated styling with `css` template literals
+- **Template Literals**: Declarative HTML templates with `html` tag
+- **Decorators**: Modern syntax with `@customElement` and `@property`
+- **TypeScript Support**: Type-safe component development
+
+#### Utrecht Design System
```

Comment:
I don't think we should mention a 1 municipality 'community' design value like this. We use our own, inheriting from the NL design-system "Estafettemodel". Sometimes that's Utrecht but definitely not always. So change Utrecht mostly into "NL Design System" here.